### PR TITLE
Port Caddy Configuration page from Wiki

### PR DESCRIPTION
**Pull request by Toakan**
Brought this page : https://github.com/pi-hole/pi-hole/wiki/Caddy-Webserver-Frontend over to be stored on the main docs site.

It's listed under the sites "Guides" section. Ran through the local mkdocs serv command to check compatibility.

image

Signed-off-by: Toakan <tdaykin@live.co.uk>

### CODE REVIEW

File: docs/guides/caddy-configuration.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Caddy Webserver Frontend
+
+If you'd like to use Caddy as your main web server with PiHole, you'll need to make a few changes.
```
Comment (Mcat12) [inline docs/guides/caddy-configuration.md]: Change "PiHole" to "Pi-hole"

File: mkdocs.yml [hunk 2]
```diff
@@ -63,6 +63,7 @@ pages:
 - 'Guides':
   - 'Pi-hole as All-Around DNS Solution': guides/unbound.md
   - 'Configuring DNS-Over-HTTPS on Pi-hole': guides/dns-over-https.md
+  - 'Configuring Caddy for Pi-Hole': guides/caddy-configuration.md
```
Comment (Mcat12) [inline mkdocs.yml]: Change "Pi-Hole" to "Pi-hole"

File: docs/guides/caddy-configuration.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
+# Caddy Webserver Frontend
```
Comment (Mcat12) [inline docs/guides/caddy-configuration.md]: Remove this header, as it will be automatically replaced with the page title in the mkdocs config.

**Comment by Toakan** [comment]
Files updated with the changes requested @Mcat12

**Pull request merged**